src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -100,7 +101,15 @@ public RestoreCommand()
 
                 Console.PrintPackageSources(packageSources);
 
-                var localCache = new NuGetv3LocalRepository(packagesDir);
+                var sharedCache = new RestoreCommandSharedCache();
+                sharedCache.LocalCache = new NuGetv3LocalRepository(packagesDir);
```

Comment:
I would make it in the constructor, and not auto create one in the restore command, but force to pass it in.Is there a scenario where the restore command needs to create its own cache?